tempesta_fw/http_parser.c

```diff
@@ -2468,61 +2468,69 @@ __year_day_secs(unsigned int year, unsigned int day_sec)
 	return (days - EPOCH_DAYS) * SEC24H + day_sec;
 }
 
-static size_t
-__skip_weekday(unsigned char *p, size_t len)
-{
-	unsigned char lc, *c, *end = p + len;
-
-	for (c = p; c < end; ++c) {
-		lc = TFW_LC(*c);
-		if ((unsigned)(lc - 'a') > (unsigned)('z' - 'a') && lc != ',')
-			return (*c != ' ') ? CSTR_NEQ : c - p;
-	}
-	return len;
-}
-
 static int
 __parse_http_date(TfwHttpMsg *hm, unsigned char *data, size_t len)
 {
-	static const unsigned long colon_a[] ____cacheline_aligned = {
-		/* ':' (0x3a)(58) Colon */
-		0x0400000000000000UL, 0, 0, 0
-	};
 	int r = CSTR_NEQ;
 	__FSM_DECLARE_VARS(hm);
 
 	__FSM_START_ALT(parser->_i_st);
 
-	__FSM_STATE(I_Date) {
-		/*
-		 * Skip a weekday with comma (e.g. "Sun,") as redundant
-		 * information.
-		 */
-		__fsm_sz = __data_remain(p);
-		__fsm_n = __skip_weekday(p, __fsm_sz);
-		if (__fsm_sz == __fsm_n)
-			__FSM_I_MOVE_n(I_Date, __fsm_sz);
-		if (unlikely(__fsm_n == CSTR_NEQ))
+	/*
+	 * Skip a weekday with comma (e.g. "Sun,") as redundant
+	 * information.
+	 */
+	__FSM_STATE(I_WDate1) {
+		unsigned char lc = TFW_LC(c);
```

Comment:
The [RFC 7231 section  Date/Time Formats](https://tools.ietf.org/html/rfc7231#section-7.1.1.1) talks about three formats, but only `IMF-fixdate` is preferred:
```
   An example of the preferred format is

     Sun, 06 Nov 1994 08:49:37 GMT    ; IMF-fixdate

   Examples of the two obsolete formats are

     Sunday, 06-Nov-94 08:49:37 GMT   ; obsolete RFC 850 format
     Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
```
In the same time `A recipient that parses a timestamp value in an HTTP header field MUST accept all three HTTP-date formats.` The proposed solution parses only ` IMF-fixdate`, but we have to support all of them, so extra steps are required.